content/reference/routing.md

```diff
@@ -0,0 +1,406 @@
+var app = choo()
+app.use((state, emitter) => {            // 1.
+  emitter.on('navigate', (route) => {    // 2.
+    console.log(`Navigated to ${route}`) // 3.
+  })
+})
+```
+
+1. Create a new store.
+2. Listen to a navigate event.
+3. Whenever navigate is emitted, we log out what the new route is we're
+   navigating to. Route is a string here.
+
+## Hash Routing
+Sometimes when you deploy a static app, you can't control the server
+part of it. Changing the route might mean the server interprets it
+differently, causing problems to occur when reloading the page.
+
+To work around this, Choo supports hash routing. Instead of writing a
+route as `foobar.com/bin/baz`, you can write it as `foobar.com#bin/baz`.
+
+```js
+var choo = require('choo')
+var app = choo()
+
+app.route('/', view)
+app.route('#hi', view)
+app.route('#hi/hello', view)
+
+function view (state, emit) {
+  return html`
+    <body>${state.route}</body>
+  `
+}
+```
+
+## Page Anchors
+Another use of hashes in urls is to map to anchors on the page. This is
+commonly used for headings in articles. So when a link is shared,
+they're navigated to the right heading in the page.
+
+Choo supports page anchors out of the box. It tries to match anchors on
+the page first. If no matching anchor is found, Choo will try to find a
+matching route in the router. If no matching route is found, the regular
+fallback behavior occurs.
+
+## Disabling Routing
+There are cases where you might not need routing at all, for example
+when using Choo to create iframe widgets. Choo accepts options in the
+contructor to disable either hrefs or the entire history API
+integration (including hrefs).
+
+Because there is no more routing events available, you'll need to handle
+it yourself. You can manually change the value of `state.href`, before
+calling the `'render'` event. Or perhaps even simpler would be to only
+use a single route. It's up to you at this point.
+
+```js
+var choo = require('choo')
+var app = choo({ href: false }) // 1.
+```
+1. Disable the handling of `<a href="">` links.
+
+
+```js
+var choo = require('choo')
+var app = choo({ history: false }) // 1.
+```
+1. Disable the handling of `<a href="">` links, and all built-in history
+   events.
+
+## Scroll Restoration
+Browsers are generally quite good at scrolling pages. When using the
+history API, previous locations on the page are restored. But this is
+not great for every app.
+
+If for example you want to always scroll to the top of the page whenever
+we navigate, you'll notice the page jumping around. This is not great
+user experience.
+
+In order to manually take control of scrolling whenever page navigation
+occurs, we can make use of `history.scrollRestoration`. Setting the
+value to `'manual'` disables all default scroll behavior, allowing us to
+take control instead.
+
+```js
+if (typeof window !== 'undefined' && window.history.scrollRestoration) {
+  window.history.scrollRestoration = 'manual'
+}
+```
+
+### Further Reading
```

Comment:
this link can be added directly to the first occurrence of `history.scollRestoration` (i.e. in line 395) imo